broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js

```diff
@@ -519,6 +519,136 @@ class ServiceBrokerApiController extends FabrikBaseController {
+            }
+            throw new UnprocessableEntity(`Service Instance cannot be fetched:lastOperation: ${lastOperation}`, 'ConcurrencyError');
+          } else if(resourceState === CONST.APISERVER.RESOURCE_STATE.SUCCEEDED || resourceState === CONST.APISERVER.RESOURCE_STATE.FAILED) {
+            // return response with 200
+            const body = {};
+            // generate dashboard client
+            const context = {};
+            const plan = catalog.getPlan(_.get(resource, 'spec.planId'));
+            _.set(context,'plan',plan);
+            _.set(context,'instance_id',req.params.instance_id);
+            const dashboardUrl = this.getDashboardUrl(context);
+            if (dashboardUrl) {
+              body.dashboard_url = dashboardUrl;
+            }
+            _.set(body,'service_id',_.get(resource, 'spec.serviceId'));
+            _.set(body,'plan_id',_.get(resource, 'spec.planId'));
+            _.set(body,'parameters',_.get(resource, 'spec.parameters'));
+
+            return res.status(CONST.HTTP_STATUS_CODE.OK).send(body);
+          }
+        })
+        .catch(BadRequest, badRequest)
+        .catch(NotFound, notFound)
+        .catch(UnprocessableEntity, unprocessableEntity);
+
+    });
+
+  }
+
+  getServiceBinding(req, res) {
+    function badRequest(err) {
+      res.status(CONST.HTTP_STATUS_CODE.BAD_REQUEST).send({});
+    }
+    function notFound(err) {
+      res.status(CONST.HTTP_STATUS_CODE.NOT_FOUND).send({});
+    }
+    function done(bindResponse, body) {
+      const response = decodeBase64(bindResponse);
+      _.merge(body,response);
+      res.status(CONST.HTTP_STATUS_CODE.OK).send(body);
+    }
+
+    return Promise.try(() => {
+      return eventmesh.apiServerClient.getResource({
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS,
+        resourceId: req.params.binding_id,
+        namespaceId: eventmesh.apiServerClient.getNamespaceId(req.params.instance_id)
+      })
+        .then(resource => {
+          const isServiceBindingRetrievable = _.get(catalog.getService(_.get(resource, 'spec.serviceId')), 'bindings_retrievable',false);
+          const isServiceBindable = _.get(catalog.getService(_.get(resource, 'spec.serviceId')), 'bindable',false);
+          // if service binding is not retrievable
+          if(!isServiceBindingRetrievable) {
+            throw new BadRequest('Service does not support binding retrieval');
+          }
+          const isPlanBindable = _.get(catalog.getPlan(_.get(resource, 'spec.planId')), 'bindable',isServiceBindable);
+          // if plan is bindable
+          if(!isPlanBindable) {
+            throw new BadRequest('Service plan is not bindable');
+          }
+          const resourceStatus = _.get(resource, 'status');
+          const resourceState = resourceStatus.state;
```

Comment:
This can be done in single `_.get` call

Reply:
@jayeshjk Done. [(Commit)](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/1239/commits/6c863f9d5ddda71cf052b9a38c70b31e870421ba)